olive/engine/engine.py

```diff
@@ -263,82 +478,19 @@ def run(
-                ):
-                    error_msg = "Azure ML model only supports AzureMLSystem for Olive Pass"
-                    logger.error(error_msg)
-                    raise Exception(error_msg)
-
-                model, model_id = self._run_pass(pass_id, pass_search_point, model, model_id, verbose)
-                if model == PRUNED_CONFIG:
-                    should_prune = True
-                    logger.info("Pruned")
-                    break
-                model_ids.append(model_id)
-
-            signal = {}
-            if not should_prune:
-                # evaluate the model
-                try:
-                    evaluator = self.evaluator_for_pass(pass_id)
-                    if self.no_search and evaluator is None:
-                        # skip evaluation if no search and no evaluator
-                        signal = None
-                    else:
-                        signal = self._evaluate_model(model, model_id, evaluator, verbose)
-                except Exception as e:
-                    logger.error(f"Evaluation failed: {e}")
-                    raise e
-                if verbose:
-                    logger.info(f"Signal: {signal}")
-
-            # there is only one step if no search
-            if self.no_search:
-                break
+            should_prune, signal, model_ids = self.run_passes(
+                next_step["passes"], model, model_id, accelerator_spec, verbose
+            )
 
             # record feedback signal
             self.search_strategy.record_feedback_signal(next_step["search_point"], signal, model_ids, should_prune)
 
             time_diff = time.time() - start_time
             self.search_strategy.check_exit_criteria(iter_num, time_diff, signal)
 
-        if self.no_search:
-            # save the model to output_dir
-            output_model_name = f"{prefix_output_name}model"
-            output_model_json = cache_utils.save_model(model_id, output_dir, output_model_name, self._config.cache_dir)
-
-            # save the evaluation results to output_dir
-            result_name = f"{prefix_output_name}metrics"
-            results_path = output_dir / f"{result_name}.json"
-            if signal is not None:
-                json.dump(signal, open(results_path, "w"), indent=4)
-
-            output = {"model": output_model_json}
-            if signal is not None:
-                output["metrics"] = signal
-
-            # Package output model as artifacts if no search
-            if packaging_config:
-                logger.info("Package output model as artifacts")
-                generate_output_artifacts(
-                    packaging_config, self.footprints, self.footprints.get_last_node(), output_dir
-                )
-
-            return output
-
-        self.footprints.to_file(output_dir / f"{prefix_output_name}footprints.json")
+        self.footprints[accelerator_spec].to_file(output_dir / f"{prefix_output_name}footprints.json")
```

Comment:
Please take `accelerator_spec` into account when generate the `prefix_output_name`.